src/Modules/Settings/Dnn.PersonaBar.Prompt/Services/CommandController.cs

```diff
@@ -16,20 +16,61 @@
 using DotNetNuke.Services.Exceptions;
 using DotNetNuke.Services.Localization;
 using DotNetNuke.Services.Log.EventLog;
+using DotNetNuke.Entities.Portals;
 
 namespace Dnn.PersonaBar.Prompt.Services
 {
     [MenuPermission(MenuName = "Dnn.Prompt")]
     [RequireHost]
-    public class CommandController : ControllerBase
+    public class CommandController : ControllerBase, IServiceRouteMapper
     {
         private static readonly ILog Logger = LoggerSource.Instance.GetLogger(typeof(CommandController));
         private static readonly string[] BlackList = { "smtppassword", "password", "pwd", "pass", "apikey" };
 
-        [HttpGet]
-        public HttpResponseMessage List()
+        private int portalId = -1;
+        private new int PortalId
         {
-            return Request.CreateResponse(HttpStatusCode.OK, CommandRepository.Instance.GetCommands().Values);
+            get
+            {
+                if (portalId == -1)
+                {
+                    portalId = base.PortalId;
+                }
+                return portalId;
+            }
+            set
+            {
+                portalId = value;
+            }
+        }
+        private PortalSettings portalSettings;
```

Comment:
I suggest naming `portalSettings` to `_portalSettings` to differentiate from passed parameter and to follow conventions.

Reply:
Sure. Feel free. Or do I need to do that?

Reply:
We have a convention that the person who reviews shouldn't do any changes in the code and leave it to the PR submitter. Therefore, I prefer that you do these changes and I will review and approve.